src/main/service/subscriptionService.ts

```diff
@@ -66,4 +73,119 @@ export class SubscriptionService {
     }
   }
 
+  public async handleNewSubscription(pendingSubscription, user): Promise<void> {
+    const selectionList = Object.keys(pendingSubscription);
+    for (const selectionName of selectionList) {
+      let hearingIdsList = [];
+      let courtIdsList = [];
+      let caseDetailsList: Hearing[] = [];
+      let courtDetailsList: Court[] = [];
+      switch (selectionName) {
+        case 'case-number':
+        case 'hearing-selections[]':
+          // pendingSubscription.selectionName gives undefined
+          Array.isArray(pendingSubscription[`${selectionName}`]) ?
+            hearingIdsList = pendingSubscription[`${selectionName}`] :
+            hearingIdsList.push(pendingSubscription[`${selectionName}`]);
+          caseDetailsList = await this.getCaseDetails(hearingIdsList);
+          // set results into cache
+          await this.setPendingSubscriptions(caseDetailsList, 'cases', user.id);
+          break;
+        case 'court-selections[]':
+          Array.isArray(pendingSubscription[`${selectionName}`]) ?
+            courtIdsList = pendingSubscription[`${selectionName}`] :
+            courtIdsList.push(pendingSubscription[`${selectionName}`]);
+          courtDetailsList = await this.getCourtDetails(courtIdsList);
+          // set results into cache
+          await this.setPendingSubscriptions(courtDetailsList, 'courts', user.id);
+          break;
+      }
+    }
+  }
+
+  public async getCaseDetails(cases): Promise<Hearing[]> {
+    const casesList = [];
+    for (const caseNumber of cases) {
+      const caseDetails = await hearingService.getCaseByNumber(caseNumber);
+      if (caseDetails) {
+        casesList.push(caseDetails);
+      }
+    }
+    return casesList;
+  }
+
+  public async getCourtDetails(courts): Promise<Court[]> {
+    const courtsList = [];
+    for (const courtId of courts) {
+      const courtDetails = await courtService.getCourtById(courtId);
+      if (courtDetails) {
+        courtsList.push(courtDetails);
+      }
+    }
+    return courtsList;
+  }
+
+  public async setPendingSubscriptions(subscriptions, subscriptionType, userId): Promise<void> {
+    await pendingSubscriptionsFromCache.setPendingSubscriptions(subscriptions, subscriptionType, userId);
+  }
+
+  public async getPendingSubscriptions(userId, subscriptionType): Promise<any[]> {
+    return await pendingSubscriptionsFromCache.getPendingSubscriptions(userId, subscriptionType);
+  }
+
+  public async subscribe(userId): Promise<boolean> {
+    let subscribed = true;
+    const casesType = 'cases';
+    const courtsType = 'courts';
```

Comment:
these can live at class level?

Reply:
It is only used within that function so I thought it might be better to keep it scoped?